sklearn/utils/__init__.py

```diff
@@ -79,593 +71,14 @@
     "safe_mask",
     "gen_batches",
     "gen_even_slices",
+    "_safe_indexing",
```

Comment:
Do we really want to add it here?

Reply:
Instead we could just have a:

```python
# Make _safe_indexing importable from here for backward compat as this particular
# helper is considered semi-private and typically very useful for third-party
# libraries that want to comply with scikit-learn's estimator API. In particular,
# _safe_indexing was included in our public API documentation despite the leading
# `_` in its name.
from ._indexing import _safe_indexing  # noqa
```

Reply:
Although I am not 100% sure what makes `_safe_indexing` so special compared to  `_get_column_indices` or `_safe_assign` for instance.

Reply:
This is the kind of question we'll have to answer when we start the work on the dev api

Reply:
`_safe_indexing` was initially public and became private and we wanted to make the migration easy. i.e. `utils.safe_indexing` -> `utils._safe_indexing`.

`_get_column_indices` and `_safe_assign` started off private.